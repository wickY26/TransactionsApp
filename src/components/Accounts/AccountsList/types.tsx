import { Account } from '../../../containers/Accounts/types';

export interface AccountsListProps {
  accounts: Account[];
};
