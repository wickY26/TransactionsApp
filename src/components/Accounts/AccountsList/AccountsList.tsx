import Avatar from '@material-ui/core/Avatar';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemAvatar from '@material-ui/core/ListItemAvatar';
import ListItemText from '@material-ui/core/ListItemText';
import AccountBoxIcon from '@material-ui/icons/AccountBox';
import React, { FC } from 'react';

import { AccountsListProps } from './types';

const AccountsList: FC<AccountsListProps> = ({ accounts }) => (
  <List disablePadding>
    {
      accounts.map(({ name, surname, id }) => (
        <ListItem key={id}>
          <ListItemAvatar>
            <Avatar>
              <AccountBoxIcon />
            </Avatar>
          </ListItemAvatar>
          <ListItemText
            primary={`${name} ${surname}`}
            secondary="Secondary text"
          />
        </ListItem>
      ))
    }
  </List>
);

export default AccountsList;