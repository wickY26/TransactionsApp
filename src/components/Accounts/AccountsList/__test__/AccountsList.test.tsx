import React from 'react';
import { render } from '@testing-library/react';

import AccountsList from '..';

const mockAccounts = [{name:'Poyraz', surname: 'Yilmaz', id: 0}];

const renderComponent = () => render(
  <AccountsList accounts={mockAccounts} />
);

describe('AccountsList', () => {

  it('should render correctly', () => {
    const component = renderComponent();
    expect(component.asFragment()).toMatchSnapshot();
  });

});
