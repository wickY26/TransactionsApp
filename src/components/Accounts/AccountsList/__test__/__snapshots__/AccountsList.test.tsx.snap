// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountsList should render correctly 1`] = `
<DocumentFragment>
  <ul
    class="MuiList-root"
  >
    <li
      class="MuiListItem-root MuiListItem-gutters"
    >
      <div
        class="MuiListItemAvatar-root"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 5v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2H5c-1.11 0-2 .9-2 2zm12 4c0 1.66-1.34 3-3 3s-3-1.34-3-3 1.34-3 3-3 3 1.34 3 3zm-9 8c0-2 4-3.1 6-3.1s6 1.1 6 3.1v1H6v-1z"
            />
          </svg>
        </div>
      </div>
      <div
        class="MuiListItemText-root MuiListItemText-multiline"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          Poyraz Yilmaz
        </span>
        <p
          class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
        >
          Secondary text
        </p>
      </div>
    </li>
  </ul>
</DocumentFragment>
`;
